SEP-0012.md

```diff
@@ -0,0 +1,495 @@
+# SEP-0012 -- NDCube 2
+
+| SEP           | num                                                     |
+|---------------|---------------------------------------------------------|
+| title         | NDCube 2                                                |
+| author(s)     | [Stuart Mumford](https://orcid.org/0000-0003-4217-4642) |
+| contact email | stuart@cadair.com                                       |
+| date-creation | 2020-10-13                                              |
+| type          | standard                                                |
+| discussion    |                                                         |
+| status        | discussion                                              |
+
+# Introduction
+
+This SEP proposes a redesigned API for the next major version of the ndcube package.
+The primary objective of this SEP is to gather as much community feedback on the proposed API before it is released.
+This is also a prelude to a future change, long discussed, to have the `GenericMap` class in sunpy core inherit from `NDCube`.
+This SEP should therefore also be considered as a discussion of future changes to the `Map` API, as the API proposed here would eventually (with long deprecation) be adopted by map for the functionality provided here.
+That being said, details of the migration of Map to this API are out of scope for this SEP, just a discussion of the eventual API to be adopted.
+
+## Background
+
+The creation of ndcube was originally motivated by the desire to represent and interrogate multi-dimensional solar datasets, such as those collected with the IRIS satellite.
+The fundamental objective of the package's primary class, `NDCube`, is to provide a user interface to this data which would **unite the data with the coordinate information describing the data**.
+The benefit of `NDCube` is its ability to do array-like operations which also update the coordinate system.
+In addition, `NDCube` works hard to provide information to the user that describes their data based on the coordinate system information.
+`NDCube` also provides simple summary visualisation support included in the object.
+This uses matplotlib to produce quick, coordinate-aware, and interactive visualisations of the data.
+The final major component added in `NDCube` is the concept of "extra coords" which allows the specification and integration with coordinates describing the data not specified by the primary WCS for the cube.
```

Comment:
This should have a concrete example of where extra coords are used, similar to my suggestion above for a concrete example of the type of data NDCube is motivated by.